rcl/include/rcl/arguments.h

```diff
@@ -0,0 +1,152 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__ARGUMENTS_H_
+#define RCL__ARGUMENTS_H_
+
+#include "rcl/allocator.h"
+#include "rcl/macros.h"
+#include "rcl/types.h"
+#include "rcl/visibility_control.h"
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+struct rcl_arguments_impl_t;
```

Comment:
`rcl_command_line_arguments_t` is a pretty long name. How about `rcl_cli_arguments_t`?

Reply:
`rcl_cli_arguments_t` sounds great to me, but after thinking about it, I'm not really concerned about the name. I think in context it will be relatively clear. So I'll leave it up to you to decide what to do about it.

Reply:
I'll leave this as is because I think the name is clear enough, but I'm open to changing it in another PR.